### PR TITLE
[AMORO-2960] Support paimon s3 based catalog

### LINKED ISSUE #2960: [Feature]: Support paimon s3 based catalog
### Description

Apache Paimon can manage its own catalog in s3, we can support paimon s3 based catalog in amoro project.

### Use case/motivation

_No response_

### Describe the solution

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xleoken**
## Why are the changes needed?

Close #2960.

## Brief change log

Support paimon s3 based catalog

image

image

image

image

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-core/pom.xml [hunk 1]
```diff
@@ -140,6 +140,11 @@
             <artifactId>paimon-bundle</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.paimon</groupId>
+            <artifactId>paimon-s3</artifactId>
+        </dependency>
+
```
Comment (baiyangtx) [inline amoro-core/pom.xml:147]: I think this dependency should belond to `amoro-ams-server` module .

Comment (xleoken) [reply amoro-core/pom.xml:147]: ok

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contribution.

**Review by zhoujinsong — COMMENTED** [review]
Thanks for the contribution, I left a comment, PTAL.

File: amoro-core/src/main/java/org/apache/amoro/formats/paimon/PaimonCatalogFactory.java [hunk 2]
```diff
@@ -49,8 +49,18 @@ public PaimonCatalog create(
         url ->
             catalogProperties.put(
                 HiveCatalogOptions.HIVE_CONF_DIR.key(), new File(url.getPath()).getParent()));
-    Catalog catalog = paimonCatalog(catalogProperties, metaStore.getConfiguration());
-    return new PaimonCatalog(catalog, name);
+
+    if (CatalogMetaProperties.AUTH_CONFIGS_VALUE_TYPE_AK_SK.equalsIgnoreCase(
+        metaStore.getAuthMethod())) {
+      // s3.access-key, s3.secret-key
+      catalogProperties.put("s3.access-key", metaStore.getAccessKey());
```
[the remark this answers is not in the document]
Comment (xleoken) [reply amoro-core/src/main/java/org/apache/amoro/formats/paimon/PaimonCatalogFactory.java:56]: right, paimon doesn't define these values. 

image

[the remark this answers is not in the document]
Comment (xleoken) [reply amoro-core/src/main/java/org/apache/amoro/formats/paimon/PaimonCatalogFactory.java]: Updated to define paimon s3 keys.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**